### ISSUE TITLE
Resource.dispatch() only allows `HttpResponse`

### ISSUE THREAD

**Issue by gbansaghi** [django-tastypie/django-tastypie#1593]
The `dispatch()` method in `tastypie.resources.Resource` performs an [`isinstance` check][isinstance] that I believe is too restrictive:

```py
# If what comes back isn't a ``HttpResponse``, assume that the
# request was accepted and that some action occurred. This also
# prevents Django from freaking out.
if not isinstance(response, HttpResponse):
    return http.HttpNoContent()
```

The issue is that `HttpResponse` is not the base class for _all_ other response classes. It for most of them (e.g. `JsonResponse`, `HttpResponseNotFound`, `HttpResponseForbidden` etc.), but notably `StreamingHttpResponse` inherits only from `HttpResponseBase` (and `FileResponse` inherits from `StreamingHttpResponse`). Thus, streaming or file responses will always generate a 204 No Content.

I believe the solution might be as simple as changing this (and maybe the two other checks in this file) to `isinstance(response, HttpResponseBase)`, and I'd be happy to create a pull request if that's all it is. I wanted to open an issue first, though, to get some feedback and see if doing this might break things.

[isinstance]: https://github.com/django-tastypie/django-tastypie/blob/master/tastypie/resources.py#L506

**Comment by georgedorn** [comment]
Yeah, it seems likely that this was written prior to the existence of `HttpResponseBase`.  A PR for it would be helpful, especially if you could include a test of something like `FileResponse`.
